src/common/base/byte_utils.h

```diff
@@ -53,6 +53,48 @@ T ReverseBytes(const T* x) {
   return y;
 }
 
+template <typename T, typename TCharType = char, size_t N = sizeof(T),
+          typename = std::enable_if_t<std::is_unsigned<T>::value>>
+T LEndianBytesToUintInternal(std::basic_string_view<TCharType> buf) {
+  // Doesn't make sense to process more bytes than the destination type.
+  // Less bytes is okay, on the other hand, since the value will still fit.
+  static_assert(N <= sizeof(T));
+
+  // If source buffer doesn't have enough bytes, cap end of buffer and
+  // allow data type promotion to occur (i.e. encode uint32_t into uint64_t).
+  size_t max = N;
+  if (buf.size() < max) {
+    max = buf.size();
+  }
```

Comment:
I was trying to prevent the big and little endian 32 and 64 bit use cases from having significant branching. I originally didn't think using the existing function would work because the `N` must be known at compile time, but we really only have two `N`s -- 4 and 8 bytes.

[Here](https://github.com/pixie-io/pixie/commit/ae5dd209520f4d25520b1ed41687a7f9daf85b1e) is an alternative approach. This still requires one "extra" [CreateStringView call](https://github.com/pixie-io/pixie/commit/ae5dd209520f4d25520b1ed41687a7f9daf85b1e#diff-a5cd29c11ef9bc8c2ae92c1eb15bdace92d54e4e634fe5889a64979bd272e0b8R107) -- this PR was created to avoid needing to create a string view per byte to int conversion (of which there are 3).

I think that approach seems better than the one I linked in my Test plan, but let me know what you think.

Reply:
IMO the alternative approach you suggested seems better. I think making the caller a bit more complex at the cost of keeping the core basic API self-consistent and cleaner is nicer in the long run.

Reply:
Sounds good. I'll open up a new PR with the necessary changes for that pulled for [src/common/base/byte_utils.h](https://github.com/pixie-io/pixie/pull/1590/files/75f863f93e9f140e367a4ae440481be258da153a#diff-514144ee0d7b937393152c0457f1ae62f8ff9a2a137843023b36fd3c8d738dde)